src/backend/db/database_manager.py

```diff
@@ -850,5 +866,363 @@ def delete_ticket(self, ticket_id, customer_guid):
+                text("SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = :db_name"),
+                {"db_name": customer_db_name}
+            ).fetchone()
+
+            logger.debug(f"Database existence check result: {result}")
+            if not result:
+                raise ValueError(f"Database {customer_db_name} does not exist")
+
+            # Switch to the customer database
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Check if the ticket_id exists in the tickets table
+            logger.debug(f"Checking existence of ticket_id: {ticket_id}")
+            ticket_exists = session.execute(
+                text("SELECT 1 FROM tickets WHERE ticket_id = :ticket_id LIMIT 1"),
+                {"ticket_id": ticket_id}
+            ).fetchone()
+
+            logger.debug(f"Ticket existence check result: {ticket_exists}")
+            if not ticket_exists:
+                raise ValueError(f"Invalid ticket_id: {ticket_id} does not exist.")
+
+            # Begin transaction
+            logger.debug("Starting transaction for comment creation")
+            with session.begin_nested():
+                comment_uuid=str(uuid.uuid4())
+                # Insert into ticket_comments table
+                query = '''INSERT INTO ticket_comments (ticket_id, posted_by, comment, comment_uuid)
+                           VALUES (:ticket_id, :posted_by, :comment, :comment_uuid)'''
+                session.execute(
+                    text(query),
+                    {
+                        "ticket_id": ticket_id,
+                        "posted_by": posted_by,
+                        "comment": comment,
+                        "comment_uuid":comment_uuid
+                    },
+                )
+
+                logger.debug("Insert into ticket_comments executed")
+
+                # Fetch the generated comment_id
+                result = session.execute(
+                    text("""
+                        SELECT comment_id 
+                        FROM ticket_comments 
+                        WHERE ticket_id = :ticket_id AND comment_uuid = :comment_uuid AND posted_by = :posted_by
+                    """),
+                    {"ticket_id": ticket_id, "comment_uuid": comment_uuid, "posted_by": posted_by}
+                ).fetchone()
+
+                logger.debug(f"Fetch newly created comment_id result: {result}")
+
+                if result:
+                    comment_id = result[0]
+                    logger.debug(f"Created comment with ID: {comment_id}")
+                else:
+                    raise Exception("Failed to retrieve the newly created comment ID.")
+
+            session.commit()
+            logger.debug(f"Transaction committed successfully for comment ID {comment_id}")
+
+            return {"comment_id": comment_id, "comment": comment}
```

Comment:
The output should have

CREATE TABLE IF NOT EXISTS ticket_comments (
comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
ticket_id BIGINT NOT NULL,
posted_by VARCHAR(255) NOT NULL,
comment TEXT NOT NULL,
is_edited BOOLEAN DEFAULT FALSE,
created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
);

Reply:
modified sir